src/core/xml.cpp

```diff
@@ -1095,6 +1102,10 @@ static Task *instantiate_node(XMLParseContext &ctx,
         }
 
         try {
+            #if defined(MI_ENABLE_CUDA) || defined(MI_ENABLE_LLVM)
+            if (ctx.is_jit() && ctx.parallel)
+                jit_new_scope((JitBackend) ctx.backend);
+            #endif
```

Comment:
I am wondering whether we should move the `jit_new_scope` in the `PluginManager::create_object()` function. Even when instantiating objects sequentially, it would be safer if all plugins have their own JIT scope. 🤔

Reply:
Sure, why not. By the way, a call to `jit_new_scope` is now mandatory whenever there is a parallel code block (which potentially calls `jit_new_scope`) followed by further computation on the main thread that references the variables created in the parallel region. Otherwise the order in the generated program won't make sense.